### ISSUE TITLE
Incorrect upcasting of signed integers for comparsion

### ISSUE THREAD

**Issue by nventuro** [noir-lang/noir#5227]
### Aim

After quite a bit of head-scratching I noticed certain signed integers seem to be incorrectly upcast when performing comparisons.

### Bug

I expected the following test to pass, but it doesn't:

```rust
#[test]
fn test() {
    let a = 5;
    let b = -1;

    assert(a > b);
}
```

The test does pass as expected if I do `a as i32`.

**Comment by jfecher** [comment]
Looks like `a` and `b` are defaulting to `u64`s since `>` was used (otherwise they'd default to Fields). The `-1` then presumably overflowed the `u64`'s range.

**Comment by asterite** [comment]
Fixed by #5375

Now the above program gives this error:

```
[one] Running 1 test function
[one] Testing test... error: The value `-1` cannot fit into `u32` which has range `0..=4294967295`
  ┌─ src/main.nr:4:13
  │
4 │     let b = -1;
  │             --
```
